plugins/woocommerce/src/Utilities/PluginUtil.php

```diff
@@ -117,6 +117,39 @@ public function is_woocommerce_aware_plugin( $plugin_file_or_data ): bool {
 		}
 	}
 
+	/**
+	 * Match plugin identifier passed as a parameter with the output from `get_plugins()`.
+	 *
+	 * @param string $plugin_file Plugin identifier, either 'my-plugin/my-plugin.php', or output from __FILE__.
+	 *
+	 * @return string|false
+	 */
+	public function get_wp_plugin_id( $plugin_file ) {
+		$wp_plugins = array_keys( $this->proxy->call_function( 'get_plugins' ) );
+
+		// Try to match plugin_basename().
+		$plugin_basename = $this->proxy->call_function( 'plugin_basename', $plugin_file );
+		if ( array_key_exists( $plugin_basename, $wp_plugins ) ){
+			return $plugin_basename;
+		}
+
+		// Try to match by the my-file.php (file name only)
+		$file_name_parts = explode( DIRECTORY_SEPARATOR, $plugin_file );
```

Comment:
We should probably replace both `\` and `/` with `DIRECTORY_SEPARATOR` before this, that way hardcoded values will also work across platforms.

Reply:
Good point, added.